service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java

```diff
@@ -2004,4 +2081,72 @@ public List<LocationsDto> getAllLocationsByCourierId(Long courierId) {
                         String.format(TARIFF_NOT_FOUND_BY_LOCATION_ID, locationsDto.getId())))))
             .collect(toList());
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @Transactional
+    public void validatePaymentFromMonoBank(MonoBankPaymentResponseDto response) {
+        String decodedOrderReference = OrderUtils.decodeOrderReference(response.getOrderReference());
+        String[] ids = decodedOrderReference.split("_");
+        Order order = orderRepository.findById(Long.valueOf(ids[0]))
+            .orElseThrow(() -> new BadRequestException(PAYMENT_VALIDATION_ERROR));
+        Payment payment = createPayment(response, order, decodedOrderReference);
+        checkPaymentStatus(response, payment, order);
+    }
+
+    private Payment createPayment(MonoBankPaymentResponseDto response,
+        Order order,
+        String decodedOrderReference) {
+        return Payment.builder()
+            .id(Long.valueOf(decodedOrderReference
+                .substring(decodedOrderReference.lastIndexOf("_") + 1)))
+            .currency("UAH")
+            .amount((long) (response.getAmount() * 100))
+            .orderStatus(OrderStatus.FORMED)
+            .responseStatus(response.getStatus())
+            .senderCellPhone(order.getUser().getRecipientPhone())
+            .maskedCard(response.getPaymentInfo().getCardNumber())
+            .cardType(response.getPaymentInfo().getPaymentSystem())
+            .responseCode(Integer.valueOf(response.getErrorCode()))
+            .responseDescription(response.getFailureReason())
+            .orderTime(response.getCreatedDate())
+            .settlementDate(convertToDateOnly(response.getModifiedDate()))
+            .fee(Long.valueOf(response.getPaymentInfo().getFee()))
+            .paymentSystem(response.getPaymentInfo().getPaymentSystem())
+            .senderEmail(order.getUser().getRecipientEmail())
+            .paymentId(response.getInvoiceId())
+            .order(order)
+            .paymentType(PaymentType.AUTO)
+            .paymentStatus(PaymentStatus.UNPAID)
+            .build();
+    }
+
+    private void checkPaymentStatus(MonoBankPaymentResponseDto response, Payment payment, Order order) {
+        if (MonoBankStatuses.SUCCESS.getName().equalsIgnoreCase(response.getStatus())) {
+            payment.setPaymentStatus(PaymentStatus.PAID);
+            order.setOrderPaymentStatus(OrderPaymentStatus.PAID);
+            paymentRepository.save(payment);
+            orderRepository.save(order);
+            eventService.save(OrderHistory.ORDER_PAID, OrderHistory.SYSTEM, order);
+            eventService.save(OrderHistory.ADD_PAYMENT_SYSTEM + payment.getPaymentId(),
+                OrderHistory.SYSTEM, order);
+        }
+        if (MonoBankStatuses.FAILURE.getName().equalsIgnoreCase(response.getStatus())
```

Comment:
what should we do if we have some other status (not `SUCCESS`, `FAILURE` or `REVERSED`)?